doc/sphinx/Pacemaker_Explained/constraints.rst

```diff
@@ -131,15 +131,19 @@ ___________________
        role
      - :ref:`enumeration <enumeration>`
      - ``Started``
-     - The constraint applies only to resource instances in this role;
-       significant only for :ref:`promotable clones <s-resource-promotable>`.
-       Allowed values:
+     - This is significant only for
+       :ref:`promotable clones <s-resource-promotable>`, is allowed only if
+       ``rsc`` or ``rsc-pattern`` is set, and is ignored if the constraint
```

Comment:
You said the same applies to score (ignored if the constraint contains a rule); not currently mentioned under score above

Reply:
The schema prevents specifying a rule and a score, so the description says "A location constraint must either have a node and score, or contain at least one rule."

The schema currently has role with the rsc/rsc-pattern vs resource_ref choice, which is separate from the node/score vs rule choice. It's certainly possible to straighten that out but I'm sick of this schema ;)